org.eclipse.xtext/src/org/eclipse/xtext/resource/EObjectAtOffsetHelper.java

```diff
@@ -157,4 +166,45 @@ protected EObject resolveCrossReferencedElement(INode node) {
 		return null;
 	}
 
+	private String getUnquotedToxenText(INode node) {
+		String text = NodeModelUtils.getTokenText(node);
+		if (text != null && text.startsWith("^")) {
+			return text.substring(1);
+		}
+		return text;
+	}
+
+	/**
+	 * Find the leaf node at the given offset that defines a named element or cross-references to such an element. May
+	 * return <code>null</code> if the given offset is not valid or if such an object cannot be found at the given
+	 * offset.
+	 * 
+	 * @param xtextResource
+	 *            the {@link XtextResource} May be <code>null</code>.
+	 * @param offset
+	 *            the offset where the leaf node is to be searched.
+	 * @param condition
+	 *            a condition that the grammar element of the corresponding leaf node must met
+	 * @return the leaf node with an identifier at the given offset or <code>null</code>.
+	 * @since 2.27
+	 */
+	public EObject getElementWithNameAt(XtextResource xtextResource, int offset, Predicate<EObject> condition) {
+		if (offset >= 0 && xtextResource != null) {
+			IParseResult parseResult = xtextResource.getParseResult();
+			if (parseResult != null) {
+				ICompositeNode rootNode = parseResult.getRootNode();
+				if (rootNode != null) {
+					EObject object = resolveElementAt(xtextResource, offset);
+					String name = SimpleAttributeResolver.NAME_RESOLVER.apply(object);
+					if (name != null && !name.isEmpty()) {
+						ILeafNode leaf = NodeModelUtils.findLeafNodeAtOffset(rootNode, offset);
+						if (leaf != null && name.equals(getUnquotedToxenText(leaf)) && !tokenUtil.isWhitespaceOrCommentNode(leaf) && condition.test(leaf.getGrammarElement())) {
```

Comment:
name.equals(getUnquotedToxenText(leaf)) is not really accurate, since it will macht any node with text equal to the name of the EObject. Actually I would like a condition that tests that the leaf node "leaf" has been used to set the attribute with name "name" ob the object, but I do not know how to code it. Thus I went for the approximation.

Any help with the condition to make it test that the leaf node has been used to assign the value of a given feature?

Reply:
shoulnt ecoreutil2.findNodesForFeature or something the like help?

Reply:
I have found that method in `NodeModelUtils` (Not ECoreUtil2), I will see if that can be used. Thanks.

Reply:
@cdietrich , thanks, that worked.